extension/arcs-content-script.js

```diff
@@ -7,38 +7,34 @@
 
 // TODO(smalls) there should be a better way to detect an arcs page we can
 // inject data into.
-const isArcsPage =
+const isExtensionAppShellPage =
   document.body.getElementsByTagName('extension-app-shell').length > 0;
 
-function sendInjectArcsDataMessage() {
-  chrome.runtime.sendMessage({ method: 'loadAllEntities' }, entities => {
-    window.postMessage({ method: 'injectArcsData', entities: entities }, '*');
-  });
-}
+if (isExtensionAppShellPage) {
+  // Listen for requests to send entities.
+  window.addEventListener('message', event => {
+    console.log('arcs-page content script received event ' + event.data.method, event.data);
+    if (event.source != window || event.data.method != 'pleaseInjectArcsData') {
+      return;
+    }
 
-if (!isArcsPage) {
-  // In the common case, if we're not running an arcs instance, extract entities
-  // from the page.
-  extractEntities(document, window.location).then(results => {
-    console.log('content-script result of extractEntities', results);
-    chrome.runtime.sendMessage({
-      method: 'storePageEntities',
-      url: window.location.toString(),
-      results: results
+    chrome.runtime.sendMessage({ method: 'loadAllEntities' }, entities => {
+      console.log('arcs-extension content script received entities from extension; forwarding to extension-app-shell for injection',
```

Comment:
Run prettier on this patch? Am guessing it's not run due to column width here. I added the format-on-save to my Atom editor and it picks up the prettier config I added to the extension `package.json`. So we have that going for us, which is nice. Maybe we can convince folks to use prettier more broadly soon.

Reply:
No, I just forgot :/